File: CPP/Implementation1_thread_unsafe/move_servant.cpp
#include "move_servant.h"
void move_servant::move_to(movable* serviced, position where)
{			
	serviced->set_position(where);
}
void move_servant::move_to(movable* serviced, int dx, int dy)
{
	serviced->set_position(position(dx,dy));
}
void move_servant::move_by(movable* serviced, int dx, int dy)
{
	dx += serviced->get_position().x_position;
	dy += serviced->get_position().y_position;
	std::this_thread::sleep_for(std::chrono::seconds{2});
	serviced->set_position(position(dx, dy));
	//std::cout << serviced->get_position() << '\n';
}
void move_servant::move_by(movable* serviced, position where)
{
	where.x_position += serviced->get_position().x_position;
	where.y_position += serviced->get_position().y_position;
	
	//std::cout << where.x_position << ": " << where.y_position << '\n';
	std::this_thread::sleep_for(std::chrono::seconds{2});
	serviced->set_position(position(where.x_position, where.y_position));
	
	//std::cout << serviced->get_position() << '\n';
}

File: JAVA/Implementation2/MoveServant.java
public class MoveServant {
	
	public void moveTo(Movable serviced, Position where) {
		serviced.setPosition(where);
	}

	public void moveTo(Movable serviced, int dx, int dy)  {
		serviced.setPosition(new Position(dx,dy));
	}	

	public void moveBy(Movable serviced, int dx, int dy) {
		
		dx += serviced.getPosition().xPosition;
		dy += serviced.getPosition().yPosition;
		serviced.setPosition(new Position(dx, dy));
	}
	
	public void moveBy(Movable serviced, Position where) {
		
		where.xPosition += serviced.getPosition().xPosition;
		where.yPosition += serviced.getPosition().yPosition;
		serviced.setPosition(new Position(where.xPosition, where.yPosition));
	}
}